src/super_gradients/training/transforms/transforms.py

```diff
@@ -655,8 +655,10 @@ def __call__(self, sample: dict) -> dict:
             cp_sample = sample["additional_samples"][0]
             img, cp_labels = cp_sample["image"], cp_sample["target"]
             cp_boxes = cp_labels[:, :4]
-
-            img, cp_boxes = _mirror(img, cp_boxes, self.flip_prob)
+            if self.prob < random.random():
```

Comment:
The condition should be `if random.random() < self.prob`. The order matters:
Imagine your have self.prob = 1, which can be translated to "Always apply" that transformation.
A `random()` returns a value in range `0.0 <= X < 1.0`.
So your condition `if self.prob < random.random():` would never be a True: `if 1 < random()` which is by definition always less than one.